### PR TITLE
Basic Readibility for functions.lua

### PR DESCRIPTION
**Pull request by HebiKotei**
🎨 : ## Description

This pull request reworks some nested requirements that impede the readability of the ```functions.lua``` file. Functionality should remain exactly the same as the basic logic has not been changed. Nesting has been improved by using more guard clauses and removing else statements.

This is just a proof of concept for the moment, am unable to personally test due to my VM currently not being set up.

## Checklist

- [ ] I have personally loaded this code into an updated qbcore project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Review by Daze13 — APPROVED** [review]
I have tested this pr on my local and it works as intended

File: server/functions.lua [hunk 1]
```diff
@@ -315,18 +313,19 @@ function QBCore.Functions.AddPermission(source, permission)
...
                 ExecuteCommand(('remove_principal player.%s qbcore.%s'):format(source, v))
                 QBCore.Commands.Refresh(source)
             end
         end
+        return
+    end
+    
+    if IsPlayerAceAllowed(source, permission) then
+        ExecuteCommand(('remove_principal player.%s qbcore.%s'):format(source, permission))
+        QBCore.Commands.Refresh(source)
```
Comment (Manason) [inline server/functions.lua:328]: I think this function is less readable now.

File: server/functions.lua [hunk 2]
```diff
@@ -335,7 +334,9 @@ end
 function QBCore.Functions.HasPermission(source, permission)
     if type(permission) == "string" then
         if IsPlayerAceAllowed(source, permission) then return true end
-    elseif type(permission) == "table" then
+    end
+
+    if type(permission) == "table" then
```
Comment (Manason) [inline server/functions.lua:339]: I also think this function is less readable than before. The guard clauses only improve readability when the guarded condition is the edge case and smaller than the "normal" case.

File: server/functions.lua [hunk 3]
```diff
@@ -335,7 +334,9 @@ end
 function QBCore.Functions.HasPermission(source, permission)
     if type(permission) == "string" then
         if IsPlayerAceAllowed(source, permission) then return true end
```
Comment (Manason) [inline server/functions.lua:336]: nested if can be combined into a single if with an and clause.

**Review by Manason — CHANGES_REQUESTED** [review]
Aside from two of the functions mentioned in the review, I like the guard functions in other places.

File: server/functions.lua (hunk 2 above)
Comment (HebiKotei) [reply server/functions.lua:339]: I think for this particular case this is very debatable, this is also for extensibility, for example, if you wanted to add another requirement/check you can just do another if statement underneath and then looks far better than a series of elseif.

File: server/functions.lua (hunk 1 above)
Comment (HebiKotei) [reply server/functions.lua:328]: I was unhappy with this one as well, I'll revert it back to the original until i can find another way to tidy it up.

**Comment by ChatDisabled** [comment]
Someone else besides me that uses guard clauses? Is it possible to learn this power?

**Pull request merged**